src/VecSim/memory/vecsim_malloc.h

```diff
@@ -1,44 +1,23 @@
 #pragma once
+#include "VecSim/vec_sim_common.h"
 #include <stddef.h>
 #include <memory>
-
-#ifdef REDIS_MODULE_TARGET /* Set this when compiling your code as a module */
-#include "redismodule.h"
 #include <cstring>
-static inline void *vecsim_malloc(size_t n) { return RedisModule_Alloc(n); }
-static inline void *vecsim_calloc(size_t nelem, size_t elemsz) {
-    return RedisModule_Calloc(nelem, elemsz);
-}
-static inline void *vecsim_realloc(void *p, size_t n) { return RedisModule_Realloc(p, n); }
-static inline void vecsim_free(void *p) { RedisModule_Free(p); }
-static inline char *vecsim_strdup(const char *s) { return RedisModule_Strdup(s); }
-
-static inline char *vecsim_strndup(const char *s, size_t n) {
-    char *ret = (char *)vecsim_malloc(n + 1);
-
-    if (ret) {
-        ret[n] = '\0';
-        memcpy(ret, s, n);
-    }
-    return ret;
-}
-#else
-
-#define vecsim_malloc  malloc
-#define vecsim_free    free
-#define vecsim_calloc  calloc
-#define vecsim_realloc realloc
-#define vecsim_strdup  strdup
-#define vecsim_strndup strndup
-
-#endif
 
 struct VecSimAllocator {
+    // Allow global vecsim memory functions to access this class.
+    friend inline void *vecsim_malloc(size_t n);
+    friend inline void *vecsim_calloc(size_t nelem, size_t elemsz);
+    friend inline void *vecsim_realloc(void *p, size_t n);
+    friend inline void vecsim_free(void *p);
+    friend inline char *vecsim_strdup(const char *s);
```

Comment:
Why are those declared as `friend` if they are part of the class? They can access the inner fields of `VecSimAllocator` anyway no?